include/mls/state.h

```diff
@@ -129,14 +141,16 @@ class State
   ///
   /// General encryption and decryption
   ///
-  MLSCiphertext encrypt(const MLSPlaintext& pt);
-  MLSPlaintext decrypt(const MLSCiphertext& ct);
+  // XXX(RLB) It would be good to hide this from upper layers.  However, Session
+  // wants it to detect self-commits.  Maybe we can pass in an optional<State>
+  // next_state option into CommitOpts to pull this into State?
+  MLSMessageContentAuth unprotect(const MLSMessage& msg);
 
   ///
   /// Application encryption and decryption
   ///
-  MLSCiphertext protect(const bytes& pt);
-  bytes unprotect(const MLSCiphertext& ct);
+  MLSMessage protect_app(const bytes& pt, const MessageOpts& msg_opts);
+  bytes unprotect_app(const MLSMessage& ct);
```

Comment:
This naming is unfortunate.  Maybe we could transfer the ugliness to the private API and have the public API be nicer.